libvips/include/vips/foreign.h

```diff
@@ -308,24 +308,42 @@ void vips_foreign_load_invalidate(VipsImage *image);
 		VIPS_TYPE_FOREIGN_SAVE, VipsForeignSaveClass))
 
 /**
- * VipsSaveable:
- * @VIPS_SAVEABLE_MONO: 1 band (eg. CSV)
- * @VIPS_SAVEABLE_RGB: 1 or 3 bands (eg. PPM)
- * @VIPS_SAVEABLE_RGBA: 1, 2, 3 or 4 bands (eg. PNG)
- * @VIPS_SAVEABLE_RGBA_ONLY: 3 or 4 bands (eg. WEBP)
- * @VIPS_SAVEABLE_RGB_CMYK: 1, 3 or 4 bands (eg. JPEG)
- * @VIPS_SAVEABLE_ANY: any number of bands (eg. TIFF)
+ * VipsForeignSaveable:
+ * @VIPS_FOREIGN_SAVEABLE_MONO: 1 band
+ * @VIPS_FOREIGN_SAVEABLE_RGB: 3 bands
+ * @VIPS_FOREIGN_SAVEABLE_CMYK: 4 bands
+ * @VIPS_FOREIGN_SAVEABLE_ALPHA: an extra band
+ * @VIPS_FOREIGN_SAVEABLE_ANY: saver supports everything (eg. TIFF)
+ *
+ * The set of image types supported by a saver.
  *
  * See also: #VipsForeignSave.
  */
+typedef enum /*< flags >*/ {
+	VIPS_FOREIGN_SAVEABLE_MONO = 1,
+	VIPS_FOREIGN_SAVEABLE_RGB = 2,
+	VIPS_FOREIGN_SAVEABLE_CMYK = 4,
+	VIPS_FOREIGN_SAVEABLE_ALPHA = 8,
+	VIPS_FOREIGN_SAVEABLE_ANY = 16,
+	VIPS_FOREIGN_SAVEABLE_ALL = 31,
+} VipsForeignSaveable;
+
+/* This has been deprecated and replaced by VipsForeignSaveable.
+ */
 typedef enum {
-	VIPS_SAVEABLE_MONO,
-	VIPS_SAVEABLE_RGB,
-	VIPS_SAVEABLE_RGBA,
-	VIPS_SAVEABLE_RGBA_ONLY,
-	VIPS_SAVEABLE_RGB_CMYK,
-	VIPS_SAVEABLE_ANY,
-	VIPS_SAVEABLE_LAST
+	VIPS_SAVEABLE_MONO =
+		VIPS_FOREIGN_SAVEABLE_MONO,
+	VIPS_SAVEABLE_RGB =
+		VIPS_FOREIGN_SAVEABLE_MONO | VIPS_FOREIGN_SAVEABLE_RGB,
+	VIPS_SAVEABLE_RGBA =
+		VIPS_SAVEABLE_RGB | VIPS_FOREIGN_SAVEABLE_ALPHA,
+	VIPS_SAVEABLE_RGBA_ONLY =
+		VIPS_FOREIGN_SAVEABLE_RGB | VIPS_FOREIGN_SAVEABLE_ALPHA,
+	VIPS_SAVEABLE_RGB_CMYK =
+		VIPS_SAVEABLE_RGB | VIPS_FOREIGN_SAVEABLE_CMYK,
+	VIPS_SAVEABLE_ANY =
+		VIPS_FOREIGN_SAVEABLE_ANY,
+	VIPS_SAVEABLE_LAST = 99,
 } VipsSaveable;
```

Comment:
We could still move this to `almostdeprecated.h` and alias `vips_saveable_get_type()` to `vips_foreign_saveable_get_type()` for compat, see e.g. commit https://github.com/kleisauke/libvips/commit/865d67c2416eb6d48dfb3d9d218c326601479b91. Not sure if that's the right approach, though.

Reply:
Ah maybe that is better. It'll be ages before we can get rid of the use of `saveable_get_type` in pyvips.